Formula/glibc.rb

```diff
@@ -190,6 +239,22 @@ def post_install
     share.install_symlink sys_zoneinfo if sys_zoneinfo.exist? && !brew_zoneinfo.exist?
   end
 
+  def caveats
+    <<~EOS
+      The Homebrew's Glibc has been installed with the following executables in your PATH:
+        #{HOMEBREW_PREFIX}/bin/ldd
+        #{HOMEBREW_PREFIX}/bin/ld.so
+        #{HOMEBREW_PREFIX}/sbin/ldconfig
+
+      By default, Homebrew's linker will not search for the system's libraries. If you want Homebrew to do
+      so (at your own risk), run:
+
+        cp "#{HOMEBREW_PREFIX}/etc/ld.so.conf.d/99-system-ld.conf.keepme" "#{HOMEBREW_PREFIX}/etc/ld.so.conf.d/99-system-ld.conf"
+
+      to add the system libraries to Homebrew's ld search paths.
```

Comment:
This needs more warnings that this is not recommended, as the `at your own risk` is not very clear (and may not even be read).

I'd consider omitting this entirely.

I think we also want a `brew doctor` check that warns about a `.conf` file that reads directories outside of `HOMEBREW_PREFIX`.

Reply:
Both adding the system ld paths or not can cause problems. I think we should prompt a message to the user and let them make their own decision.

- If we add system ld paths: mixing linkage of both Homebrew and system libraries.
- If we not add system ld paths: some dependent formulae will fail to find system libraries, even if `glibc` is keg-only and not present in the user's `PATH`. For example, the `python@3` formula is built with brewed `glibc`, which makes the `ctypes` module using brewed `ld` to find libraries. See https://github.com/Homebrew/homebrew-core/pull/109998#issuecomment-1240721785. `brew unlink glibc` won't work, unless the user add the system `/etc/ld.so.conf` and rebuild `ld.so.cache`.

Reply:
> * If we not add system ld paths: some dependent formulae will fail to find system libraries

This is by design. Mixing Homebrew and system libraries is not. So we should definitely add more warnings about adding your system libraries here.